onnxruntime/core/framework/session_state.cc

```diff
@@ -252,7 +251,9 @@ Status SessionState::GeneratePatternGroupCache(const std::vector<std::reference_
             }
             len *= it->second;
           } else {
-            len *= dim.dim_value();
+            if (dim.has_dim_value()) {
```

Comment:
should we error out if neither dim_value nor dim_param is set? that means it's unknown: https://github.com/onnx/onnx/blob/master/docs/IR.md#static-tensor-shapes

Reply:
sure